Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/retention/ContactDiaryCleanTask.kt

```diff
@@ -0,0 +1,66 @@
+package de.rki.coronawarnapp.contactdiary.retention
+
+import de.rki.coronawarnapp.task.Task
+import de.rki.coronawarnapp.task.TaskFactory
+import de.rki.coronawarnapp.task.common.DefaultProgress
+import kotlinx.coroutines.channels.ConflatedBroadcastChannel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.asFlow
+import org.joda.time.Duration
+import timber.log.Timber
+import javax.inject.Inject
+import javax.inject.Provider
+
+class ContactDiaryCleanTask @Inject constructor(
+    private val retentionCalculation: ContactDiaryRetentionCalculation
+) : Task<DefaultProgress, Task.Result> {
+
+    private val internalProgress = ConflatedBroadcastChannel<DefaultProgress>()
+    override val progress: Flow<DefaultProgress> = internalProgress.asFlow()
+
+    private var isCanceled = false
+
+    override suspend fun run(arguments: Task.Arguments) = try {
+        Timber.d("Running with arguments=%s", arguments)
+
+        retentionCalculation.clearObsoleteContactDiaryLocationVisits()
+        Timber.tag(TAG).d("Obsolete contact diary location visits cleaned up")
+
+        retentionCalculation.clearObsoleteContactDiaryPersonEncounters()
+        Timber.tag(TAG).d("Obsolete contact diary person encounters cleaned up")
+
+        object : Task.Result {}
+    } catch (error: Exception) {
+        Timber.tag(TAG).e(error)
+        throw error
+    } finally {
+        Timber.i("Finished (isCanceled=$isCanceled).")
+        internalProgress.close()
+    }
+
+    override suspend fun cancel() {
+        Timber.w("cancel() called.")
+        isCanceled = true
+    }
+
+    class Config : TaskFactory.Config {
```

Comment:
Forgot to set error handling to SILENT

Reply:
https://github.com/corona-warn-app/cwa-app-android/pull/1936/files